### DISCUSSION TITLE
[Bug] if layer has optionalAny, keys should allow optionalAny as well

### DISCUSSION THREAD

**Discussion by DrakeEvans** [evan-liu/karabiner.ts#116] [Ideas]
https://github.com/evan-liu/karabiner.ts/blob/40f160ece576d3f77e003ae8679b865883b1b83e/src/config/layer.ts#L154

This code above enforces layers with modifiers do not allow keymaps to have modifiers but does not handle the case when both need to have optional any modifiers

This code produces valid configuration for example and is required to allow common shortcuts like command + left_arrow etc for the remapped keys
```
layer('semicolon', 'semicolon-down')
    .modifiers({ optional: 'any' })
    .manipulators([
      withModifier({ optional: 'any'})([
        map('l').toAfterKeyUp('right_arrow'),
        withMapper({
          j: 'left_arrow',
          k: 'down_arrow',
          l: 'right_arrow',
          i: 'up_arrow',
        })((k, v) => map(k).to(v as ToKeyCode)),
      ]),
    ]),
```

**Comment by evan-liu** [comment]
Hi @DrakeEvans , thanks for trying the project out. It is a cool idea to pass optionalAny on the layer to manipulators. I've made it the default behaviour, so you do not need to manually add `withModifier('??')`:

```typescript
layer(';', 'arrow-mode')
  .modifiers('??')
  .manipulators({
    j: toKey('←'),
    i: toKey('↑'),
    k: toKey('↓'),
    l: toKey('→'),
  }),
```

Please update (`1.20.0`) and give it a try, and let me know if it works for you.

**Comment by DrakeEvans** [comment]
I rebased and it looks like its working!! Thank you
